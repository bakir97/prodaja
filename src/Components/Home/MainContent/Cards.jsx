import React from "react";
import Card from "../../../Layout/Card/Card";
import { Grid } from "@material-ui/core";
const Cards = ({ proizvodi, layout, history }) => {
  return proizvodi.map(({ naslov, slike, cijena, tezina, _id }) => (
    <Grid key={_id} item container justify="center" xs={layout}>
      <Card
        history={history}
        id={_id}
        naslov={naslov}
        slika={slike[0]}
        cijena={cijena}
        tezina={tezina}
      />
    </Grid>
  ));
};

export default Cards;
