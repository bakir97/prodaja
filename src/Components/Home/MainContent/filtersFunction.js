export default filter => {
  if (typeof filter.minCijena === "string") {
    return { minCijena: filter.minCijena };
  }
  if (typeof filter.maxCijena === "string") {
    return { maxCijena: filter.maxCijena };
  }
  if (typeof filter.minTezina === "string") {
    return { minTezina: filter.minTezina };
  }
  if (typeof filter.maxTezina === "string") {
    return { maxTezina: filter.maxTezina };
  }
};
