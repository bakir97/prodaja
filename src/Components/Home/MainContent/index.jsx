import React, { Component } from "react";
import { connect } from "react-redux";
import withWidth from "@material-ui/core/withWidth";
import {
  getProizvodi,
  errorPostProizvod
} from "../../../redux/actions/proizvodiActions";
import MainContentLayout from "./MainContent";
class MainContent extends Component {
  componentWillUnmount() {
    this.props.errorPostProizvod(false);
  }
  componentDidMount() {
    this.props.getProizvodi();
  }
  render() {
    return <MainContentLayout {...this.props} />;
  }
}

const mapStateToProps = state => ({
  proizvodi: state.proizvodi.proizvodi,
  errorState: state.proizvodi.error
});

const mapDispatchToProps = {
  getProizvodi,
  errorPostProizvod
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withWidth()(MainContent));
