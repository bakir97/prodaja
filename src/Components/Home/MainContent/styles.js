export default {
  rootMobile: {
    height: "calc(100vh - 65px)",
    marginTop: 65
  },
  rootDesktop: {
    height: "calc(100vh - 75px)",
    marginTop: 75,
    paddingLeft: "10%",
    paddingRight: "10%"
  },
  slika: {
    height: "300px",
    width: "100%",
    maxHeight: "500px",
    maxWidth: "500px",
    borderRadius: "60%"
  }
};
