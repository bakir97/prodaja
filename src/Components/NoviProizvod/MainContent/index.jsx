import React, { Component } from "react";
import MainContentLayout from "./MainContentLayout";

export class MainContent extends Component {
  state = {
    slike: [],
    oldPictures: []
  };
  componentDidUpdate() {
    if (
      Object.keys(this.props.newProizvodState).length > 0 &&
      this.state.slike.length > 0
    ) {
      this.setState({ slike: [] });
    }
    if (
      !this.props.match.params.id &&
      Object.keys(this.props.proizvod).length > 0
    ) {
      this.props.proizvodToRedux({});
      this.setState({ slike: [], oldPictures: [] });
    }
  }
  componentWillReceiveProps(nextProps) {
    if (
      this.props.proizvod !== nextProps.proizvod &&
      Object.keys(nextProps.proizvod).length > 0
    ) {
      this.setState({ oldPictures: nextProps.proizvod.slike });
    }
  }
  drop = files => {
    this.setState(prevState => ({
      slike: [...prevState.slike, ...files]
    }));
  };
  obrisi = id => {
    this.setState(prevState => ({
      slike: prevState.slike.filter(slika => slika.preview !== id)
    }));
  };
  obrisiOldPictures = id => {
    this.setState(prevState => ({
      oldPictures: prevState.oldPictures.filter(
        oldPictures => oldPictures !== id
      )
    }));
  };
  render() {
    return (
      <MainContentLayout
        {...this.props}
        {...this.state}
        drop={this.drop}
        obrisi={this.obrisi}
        obrisiOldPictures={this.obrisiOldPictures}
      />
    );
  }
}
export default MainContent;
