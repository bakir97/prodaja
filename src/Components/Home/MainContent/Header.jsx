import React, { Component } from "react";
import filtersFunction from "./filtersFunction";
import SearchBarComponent from "../SearchBar";
import Switch from "../Switch";
import LayoutButtons from "../LayoutButtons";
import Filters from "../Filters";
import Grid from "@material-ui/core/Grid";
export default class Header extends Component {
  state = {
    filters: false,
    filter: {
      minCijena: "",
      maxCijena: "",
      minTezina: "",
      maxTezina: ""
    }
  };
  filters = filter => {
    this.setState(filtersFunction(filter));
  };
  changeFiltersState = () => {
    this.setState(prevState => ({
      filters: !prevState.filters
    }));
  };
  search = e => {
    const { minCijena, minTezina, maxCijena, maxTezina } = this.state;
    const podaci = {};
    podaci.search = e;
    podaci.minCijena = minCijena;
    podaci.maxCijena = maxCijena;
    podaci.minTezina = minTezina;
    podaci.maxTezina = maxTezina;
    this.props.getProizvodi(podaci);
  };
  render() {
    const { width, changeLayout } = this.props;
    const { filters, minCijena, maxCijena, maxTezina, minTezina } = this.state;

    return (
      <>
        <Grid container justify="center">
          <Grid item container justify="center" xs={12}>
            <SearchBarComponent search={this.search} />
          </Grid>

          <Switch
            filters={filters}
            changeFiltersState={this.changeFiltersState}
          />
          {filters && (
            <Filters
              minCijena={minCijena}
              maxCijena={maxCijena}
              minTezina={minTezina}
              maxTezina={maxTezina}
              filters={this.filters}
            />
          )}
        </Grid>
        {width === "xs" ? null : <LayoutButtons changeLayout={changeLayout} />}
      </>
    );
  }
}
