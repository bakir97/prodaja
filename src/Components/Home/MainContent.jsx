import React, { Component } from "react";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import { connect } from "react-redux";
import Slika from "../../assets/global_shopping_experience.gif";
import Card from "../../Layout/Card/Card";
import withWidth from "@material-ui/core/withWidth";
import Styles from "./styles";
import SearchBarComponent from "./SearchBar/SearchBar";
import Switch from "./Switch";
import LayoutButtons from "./LayoutButtons";
import {
  getProizvodi,
  errorPostProizvod
} from "../../redux/actions/proizvodiActions";
import SnackBar from "../../Layout/SnackBar/SnackBar";
import Filters from "./Filters";
const styles = Styles;
class MainContent extends Component {
  state = {
    filters: false,
    layout: 12,
    filter: {
      minCijena: "",
      maxCijena: "",
      minTezina: "",
      maxTezina: ""
    }
  };
  componentDidUpdate() {
    if (this.props.width === "xs" && this.state.layout !== 12) {
      this.setState({ layout: 12 });
    }
  }
  componentWillUnmount() {
    this.props.errorPostProizvod(false);
  }
  componentDidMount() {
    this.props.getProizvodi();
  }
  cijenaFilters = filter => {
    if (typeof filter.minCijena === "string") {
      return this.setState({ minCijena: filter.minCijena });
    }
    if (typeof filter.maxCijena === "string") {
      return this.setState({ maxCijena: filter.maxCijena });
    }
    if (typeof filter.minTezina === "string") {
      return this.setState({ minTezina: filter.minTezina });
    }
    if (typeof filter.maxTezina === "string") {
      return this.setState({ maxTezina: filter.maxTezina });
    }
  };
  changeFiltersState = () => {
    this.setState(prevState => ({
      filters: !prevState.filters
    }));
  };
  changeLayout = number => {
    if (number !== this.state.layout) this.setState({ layout: number });
  };
  search = e => {
    const { minCijena, minTezina, maxCijena, maxTezina } = this.state;
    const podaci = {};
    podaci.search = e;
    podaci.minCijena = minCijena;
    podaci.maxCijena = maxCijena;
    podaci.minTezina = minTezina;
    podaci.maxTezina = maxTezina;
    this.props.getProizvodi(podaci);
  };
  snackRemove = e => {
    this.props.errorPostProizvod(e);
  };
  render() {
    const { classes, width, proizvodi, errorState, history } = this.props;
    const {
      filters,
      layout,
      minCijena,
      maxCijena,
      maxTezina,
      minTezina
    } = this.state;

    return (
      <Grid
        container
        alignContent="flex-start"
        alignItems="flex-start"
        justify="center"
        className={width === "xs" ? classes.rootMobile : classes.rootDesktop}
      >
        <Grid container justify="center">
          <Grid item container justify="center" xs={12}>
            <SearchBarComponent search={this.search} />
          </Grid>

          <Switch
            filters={filters}
            changeFiltersState={this.changeFiltersState}
          />
          {filters && (
            <Filters
              minCijena={minCijena}
              maxCijena={maxCijena}
              minTezina={minTezina}
              maxTezina={maxTezina}
              cijenaFilters={this.cijenaFilters}
            />
          )}
        </Grid>
        {width === "xs" ? null : (
          <LayoutButtons changeLayout={this.changeLayout} />
        )}
        {proizvodi.length > 0 ? (
          proizvodi.map(({ naslov, slike, cijena, tezina, _id }) => (
            <Grid key={_id} item container justify="center" xs={layout}>
              <Card
                history={history}
                id={_id}
                naslov={naslov}
                slika={slike[0]}
                cijena={cijena}
                tezina={tezina}
              />
            </Grid>
          ))
        ) : (
          <img className={classes.slika} src={Slika} alt="" />
        )}
        {errorState && (
          <SnackBar
            variantSnackBar="error"
            poruka={errorState}
            snackRemove={this.snackRemove}
          />
        )}
      </Grid>
    );
  }
}

const mapStateToProps = state => ({
  proizvodi: state.proizvodi.proizvodi,
  errorState: state.proizvodi.error
});

const mapDispatchToProps = {
  getProizvodi,
  errorPostProizvod
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withWidth()(withStyles(styles)(MainContent)));
