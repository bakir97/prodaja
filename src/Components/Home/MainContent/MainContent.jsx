import React, { Component } from "react";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Slika from "../../../assets/global_shopping_experience.gif";
import Cards from "./Cards";
import Styles from "./styles";
import Header from "./Header";
import SnackBar from "../../../Layout/SnackBar/SnackBar";
const styles = Styles;
class MainContent extends Component {
  state = {
    layout: 12
  };
  componentDidUpdate() {
    if (this.props.width === "xs" && this.state.layout !== 12) {
      this.setState({ layout: 12 });
    }
  }
  changeLayout = number => {
    if (number !== this.state.layout) this.setState({ layout: number });
  };

  snackRemove = e => {
    this.props.errorPostProizvod(e);
  };
  render() {
    const {
      classes,
      width,
      proizvodi,
      errorState,
      history,
      getProizvodi
    } = this.props;
    const { layout } = this.state;

    return (
      <Grid
        container
        alignContent="flex-start"
        alignItems="flex-start"
        justify="center"
        className={width === "xs" ? classes.rootMobile : classes.rootDesktop}
      >
        <Header
          width={width}
          getProizvodi={getProizvodi}
          changeLayout={this.changeLayout}
        />
        {proizvodi.length > 0 ? (
          <Cards proizvodi={proizvodi} layout={layout} history={history} />
        ) : (
          <img className={classes.slika} src={Slika} alt="" />
        )}
        {errorState && (
          <SnackBar
            variantSnackBar="error"
            poruka={errorState}
            snackRemove={this.snackRemove}
          />
        )}
      </Grid>
    );
  }
}

export default withStyles(styles)(MainContent);
