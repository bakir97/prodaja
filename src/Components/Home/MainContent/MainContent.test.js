import MainContent from "./MainContent";
describe("MainContent", () => {
  const proizvodi = [
    {
      naslov: "test",
      slike: [1, 2],
      _id: 1,
      cijena: 1,
      tezina: 1
    },
    {
      naslov: "test2",
      slike: [1, 2],
      _id: 2,
      cijena: 2,
      tezina: 2
    }
  ];
  const wrapper = shallow(<MainContent proizvodi={proizvodi} />);
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
