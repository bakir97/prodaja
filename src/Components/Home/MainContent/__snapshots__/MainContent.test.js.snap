// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent should render 1`] = `
<MainContent
  classes={
    Object {
      "rootDesktop": "MainContent-rootDesktop-2",
      "rootMobile": "MainContent-rootMobile-1",
      "slika": "MainContent-slika-3",
    }
  }
  proizvodi={
    Array [
      Object {
        "_id": 1,
        "cijena": 1,
        "naslov": "test",
        "slike": Array [
          1,
          2,
        ],
        "tezina": 1,
      },
      Object {
        "_id": 2,
        "cijena": 2,
        "naslov": "test2",
        "slike": Array [
          1,
          2,
        ],
        "tezina": 2,
      },
    ]
  }
/>
`;
